src/fits.cc

```diff
@@ -848,10 +848,10 @@ void Fits::readTableArray(std::size_t row, int col, int nElements, T *value) {
     }
 }
 
-void Fits::readTableScalar(std::size_t row, int col, std::string &value) {
+void Fits::readTableScalar(std::size_t row, int col, std::string &value, bool isVariableLength) {
     int anynul = false;
-    long size = getTableArraySize(col);
-    // We can't directly write into a std::string (may be changing in C++11).
+    long size = isVariableLength ? getTableArraySize(row, col) : getTableArraySize(col);
```

Comment:
The `(row, col)` version calls `fits_read_descript`, which is documented to "Return the descriptor for a variable length column in a binary table..." so I don't think it is safe to use on fixed-width columns.